### PR TITLE
Database and models

### PR DESCRIPTION
**Pull request by allyjay317**
Set up api's database and access to it, in preparation for route creation
[User Story](https://trello.com/c/T9noG7kG/6-release-1-as-a-user-i-want-to-be-able-to-create-an-account-to-be-able-to-save-searches-and-preferences)
[Trello Task](https://trello.com/c/wuVFtEd6/30-setup-database)

# Migrations
## Favorites
| column | type | description |
| --- | --- | --- |
| email | sting | The email that the user used to sign up |
| zip_code | integer | zip code of the city the user favorited |

## Statistics
| column | type | description |
| --- | --- | --- |
| id | integer, unique | id for statistic type |
| name | string, unique, non nullable | name of statistic |

## User Stats
| column | type | description |
| --- | --- | --- |
| email | sting | The email that the user used to sign up |
| stat_id | integer, foreign key [statistics.id] | The statistic the user has declared that they're interested in seeing |

# Models
#### Added insert and delete functions to profile model

> addFavorite
> - params
>    - email
>    - zip_code
> - inserts a new row in the favorites table with values [email, zip_code]

> removeFavorite
> - params
>    - email
>    - zip_code
> - finds and removes row with values [email, zip_code] in [okta_email, zip_code]

> removePreferredStatistic
> - params
>    - email
>    - zip_code
> - finds and removes row in the user_stats table with values [email, zip_code]

### CODE REVIEW

**Review by songamugenzi — COMMENTED** [review]
Great job with the description detailing how to use the database, the types of accepted inputs, and expected outputs.

File: api/favorites/favoritesModel.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const db = require('../../data/db-config');
+
+const getFavoritesByEmail = async (email) => {
+  return db('favorites').where({ email });
+};
+
+const addFavorite = async (email, zip_code) => {
```
Comment (bendennett) [inline api/favorites/favoritesModel.js:7]: This is well named and sensible.

**Review by bendennett — APPROVED** [review]
The supporting documentation was apparent.  The pr was relatively small and practically atomic.

**Review by songamugenzi — APPROVED** [review]

**Comment by allyjay317** [comment]
[Setup Database](https://trello.com/c/wuVFtEd6/30-setup-database)

**Pull request merged**